rpc/network/network.go

```diff
@@ -148,7 +143,22 @@ func (nm *Manager) Init(networks []params.Network) error {
 						errors += fmt.Sprintf("error updating network fallback_url for ChainID: %d, %s", currentNetworks[j].ChainID, err.Error())
 					}
 				}
-				err := nm.UpdateRPCURLs(currentNetworks[j].ChainID, networks[i].RPCURL, networks[i].OriginalRPCURL, networks[i].FallbackURL, networks[i].OriginalFallbackURL)
+
+				if networks[i].OriginalRPCURL != currentNetworks[j].OriginalRPCURL && currentNetworks[j].RPCURL == currentNetworks[j].OriginalRPCURL {
+					err := nm.UpdateRPCURL(networks[i].ChainID, networks[i].OriginalRPCURL)
```

Comment:
@alaibe wondering if it makes sense to do these queries as transactions instead of executing each query separately? 

Cause in the current implementation if an error occurs at some point, we return the errors list but do not revert the changes.

Reply:
yes currently we only log the errors, are they really independent query or are they a group of query? good question. I am wondering if we should not return the error

Reply:
They are independent queries that are executed successively and theoretically if all of them as an entirety introduce the change then if an error occurs among any two we should roll back. That's why I proposed to make a transaction and get back as soon as we face an error.

But in practice, I don't have anything against this approach you did, since the only error which may happen in that loop is the DB error, and in reality if that happens we don't have so much to do, cause the app won't run anymore.

So you can leave it as it's now, it's already approved. :)